core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -139,47 +140,51 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
 
   private def hour(millis: Long = System.currentTimeMillis()) = millis / 1000 / 60 / 60
 
-  def recoverIndex(): Future[Unit] = {
+  def recoverIndex(): Future[Long] = {
     if (downsampleConfig.enablePersistentIndexing) {
       partKeyIndex.getCurrentIndexState() match {
         case (IndexState.Empty, _)              =>
           logger.info("Found index state empty, bootstrapping downsample index")
-          recoverIndexInternal
+          recoverIndexInternal(None)
+        case (IndexState.Synced, checkpointMillis)             =>
```

Comment:
should be called with checkpointMillis but it's not recoverIndexInternal, should be something else, for ex recoverIndexSync, internally it should do logic similar to recover index fresh. (refreshWithDownsamplePartKey)